feat: new solution doc updates & contributing updates (#2068)
Azure/Azure-Verified-Modules: docs/content/usage/concepts.md

A private registry - that is hosted in your own environment - can store modules originating from the public registry. Using a private registry still grants you the latest version of AVM modules while allowing you to review each version of each module before admitting them to your private registry. You also have control over who can access your own private registry. Note that using a private registry means that you're still using each module as is, without making any changes.

> #### Inner-sourcing (using your own private registry, repositories or other storage options)

Reviewer: While I see why you'd want to mention it in the title, I would not include a private registry in this title and would keep the focus on where the source code is kept (private repository); if you want to mention it, at least try not leading with it in the title. 

Note: the private registry scenario is mentioned above.

Inner-sourcing AVM means maintaining your own, synchronized copy of AVM modules in your own internal private registry, repositories or other storage option. Customers normally look to inner-source AVM modules when they have strict security and compliance requirements, or when they want to publish their own lightly wrapped versions of the modules to meet their specific needs; for example changing some allowed or default values for parameter or variable inputs.

This is a more complex approach and requires more effort to maintain, but it can be beneficial in certain scenarios, however, it should not be the default approach as it can lead to a lot of overhead and maintenance and requires significant skills and resources to set up and maintain.

> There are many ways to approach inner-sourcing AVM modules for both Bicep and Terraform. And today the AVM team are yet to provide any specific guidance on how to do this as they are still assessing the best approach whilst working with customers. The AVM team will be publishing guidance on this topic in the future.

Reviewer: ```suggestion
There are many ways to approach inner-sourcing AVM modules for both Bicep and Terraform. The AVM team will be publishing guidance on this topic, based on customer experience and learnings.
```
